Invalid escape sequence \Z

```
fitting.py:14: in <module>
    from s3path import PureS3Path, S3Path
.venv/lib/python3.8/site-packages/s3path/__init__.py:28: in <module>
    from .old_versions import (
E     File "/builds/project/.venv/lib/python3.8/site-packages/s3path/old_versions.py", line 76
E       new_regex_pattern += '/*\Z'
E                            ^
E   SyntaxError: invalid escape sequence \Z
```

Suggestion: mark [this](https://github.com/liormizr/s3path/blob/77868add45873495226ee96bf57e85c5ad82b104/s3path/old_versions.py#L76) as raw (r-) string: 
```
E       new_regex_pattern += r'/*\Z'
```
and the problem should be solved.

I only have it on some builds, so may depend on the python version used. All of my systems have 3.8.* but it fails only on some.

--- a/s3path/old_versions.py
+++ b/s3path/old_versions.py
@@ -73,7 +73,7 @@ class _S3Flavour(_PosixFlavour):
                 new_regex_pattern += f'{self.sep}*(?s:{part.replace("**", ".*")})'
                 continue
             new_regex_pattern += f'{self.sep}{fnmatch.translate(part)[:-2]}'
-        new_regex_pattern += '/*\Z'
+        new_regex_pattern += r'/*\Z'
         return re.compile(new_regex_pattern).fullmatch
 
 

